src/CommunityVoices/App/Api/View/DisplayError.php

```diff
@@ -23,14 +25,78 @@ protected function getError($request)
 
         return $response;
     }
-    protected function getAllErrors($request)
+
+    protected function getAllErrors($request,$lines=500)
     {
-        $errors = explode('[]',str_replace('\n','',file_get_contents('/var/www/html/log/access.log')));
-        foreach($errors as &$item) {
-            $item = array("item" => $item);
-        }
-        $response = new HttpFoundation\JsonResponse(["errors" => $errors]);
-        return $response;
+        $dates = empty($request->query->get('dateRange')) ? [false,false] : explode(" - ",$request->query->get('dateRange'));
+        $linePos = empty($request->query->get('linePos')) ? PHP_INT_MAX : intval($request->query->get('linePos'));
+        $numLinesToAdd = intval($request->query->get('numLines')) ?? 0;
+        $initialRequest = $request->query->get('firstTime');
+
+        if($initialRequest)
+            return $this->tailRead(self::ERRORSLOGPATH,$lines);
```

Comment:
Hmm... https://github.com/EnvironmentalDashboard/community-voices/blob/master/src/CommunityVoices/App/Website/Bootstrap/Dispatcher.php#L39 We can have communication between a controller and a view, so maybe the controller should do all this processing and the view just responds the result?  It just seems like it contradicts MVC, where MVC wants more dumb views, I thought.

Reply:
I see your point. How would this work if I were to do this? How would the view and the controller communicate? I looked at the pr you linked but still a bit confused

Reply:
In the controller, you return something, and it becomes the last argument to the corresponding view function.

Reply:
Fixed